Cabal/Distribution/Simple/Register.hs

```diff
@@ -407,7 +408,11 @@ generalInstalledPackageInfo adjustRelIncDirs pkg abi_hash lib lbi clbi installDi
     IPI.instantiatedWith   = componentInstantiatedWith clbi,
     IPI.sourceLibName      = libName lib,
     IPI.compatPackageKey   = componentCompatPackageKey clbi,
-    IPI.license            = license     pkg,
+    -- If GHC >= 8.4 we register with SDPX, otherwise with legacy license
+    IPI.license            =
+        if ghc84
+        then Left $ either id licenseToSPDX $ licenseRaw pkg
```

Comment:
This choice converts everything to SPDX, so in GHC-8.4 package databases will only be SPDX expressions.
Alternatively we can let `licenseRaw` to go though unmodified, or convert only "sure" things (e.g. MIT, BSD3, GPL-3, not `PublicDomain` or `OtherLicense`)

Reply:
Sounds good to me :-)